samples/AspNetCore/GrpcServiceSample/Services/BankingService.cs

```diff
@@ -0,0 +1,168 @@
+            {
+                PubsubName = "pubsub",
+                Topic = "deposit"
+            });
+            result.Subscriptions.Add(new TopicSubscription
+            {
+                PubsubName = "pubsub",
+                Topic = "withdraw"
+            });
+            return Task.FromResult(result);
+        }
+
+        /// <summary>
+        /// implement OnTopicEvent to handle deposit and withdraw event
+        /// </summary>
+        /// <param name="request"></param>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public override async Task<TopicEventResponse> OnTopicEvent(TopicEventRequest request, ServerCallContext context)
+        {
+            if (request.PubsubName == "pubsub")
+            {
+                var transaction = JsonSerializer.Deserialize<Transaction>(request.Data.ToStringUtf8(), this.jsonOptions);
+                if (request.Topic == "deposit")
+                    await Deposit(transaction, context);
+                else
+                    await Withdraw(transaction, context);
+            }
+           
+            return await Task.FromResult(default(TopicEventResponse));
+        }
+
+        /// <summary>
+        /// GetAccount
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public async Task<Account> GetAccount(GetAccountInput input, ServerCallContext context)
+        {
+            var state = await _daprClient.GetStateEntryAsync<Account>(StoreName, input.Id);
+
+            if (state.Value == null)
+            {
+                return null;
+            }
+
+            return state.Value;
+        }
+
+        /// <summary>
+        /// Deposit
+        /// </summary>
+        /// <param name="transaction"></param>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public async Task<Account> Deposit(Transaction transaction, ServerCallContext context)
+        {
+            Console.WriteLine("Enter deposit");
+            var state = await _daprClient.GetStateEntryAsync<Account>(StoreName, transaction.Id);
+            state.Value ??= new Account() { Id = transaction.Id, };
+            state.Value.Balance += transaction.Amount;
+            await state.SaveAsync();
+            return state.Value;
+        }
+
+        /// <summary>
+        /// Withdraw
+        /// </summary>
+        /// <param name="transaction"></param>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public async Task<Account> Withdraw(Transaction transaction, ServerCallContext context)
+        {
+            Console.WriteLine("Enter withdraw");
+            var state = await _daprClient.GetStateEntryAsync<Account>(StoreName, transaction.Id);
+
+            if (state.Value == null)
+            {
+                return null;
+            }
```

Comment:
Is this actually right? It seems like the right thing to do here would be to report an error.

Reply:
fixed